design/fips-compliance.md

```diff
@@ -0,0 +1,199 @@
+| Trigger `kubernetes-conjur-demo` to run after every build of `conjur-authn-k8s-client` | ~30 minutes    | Depends on wether we can utilize the LOCAL_AUTHENTICATOR environment variable. If we can then the implementation time should take ~1 day | Immediate feedback as all integration tests will  run before the end of the build | Adding tests to this project is not simple. Especially error flows.                                        | The community member will not be able to investigate a failure in case  there was one so the build will never be green (unless they ask for help from a Conjur team member).  There is also no option to add integration tests. |
+| Trigger `kubernetes-conjur-demo` to run after every master build                       | ~3 minutes     | 1 hour                                                                                                                                   | After nightly build and only after the change is merged into `master`             | Adding tests to this project is not simple. Especially error flows.                                        | The community member will still be able to contribute as the tests will not run as part of the build                                                                                                                            |
+| secrets-provider model - deploy different scenarios and check output                   | ~30 minutes    | ~5 days                                                                                                                                  | Immediate feedback as all integration tests will run before the end of the build  | - Should be easier to add tests. Adding tests to the `secrets-provider-for-k8s` is pretty straight forward | The community member will not be able to investigate a failure in case there was one so the build will never be green (unless they askfor help from a Conjur team member). There is also no option to add integration tests.    |
+
+### Decision
+
+TBD
+
+### Integration Tests
+
+Regardless to how we will run our tests, it is not optimal that we have only
+ a vanilla flow. We should add another test where in case the authenticator
+ -client fails to authenticate with Conjur we don't provide an access token
+  and the log shows `CAKC015E Login failed`.
+  
+We do not need to test different permutations of error flows (e.g host does
+not exist, host is not permitted on the authn-k8s/prod authenticator) as
+these test run in the `conjur` repository. As far as the authenticator-client
+ is concerned, the output is binary - Authentication success or failure.
+   
+## Docs
+
+TBD
+
+## Version update
+
+We should update the version of:
+  - `conjur-authn-k8s-client`
+  - `seed-fetcher`
+  
+We don't need to update the version of the `secretless-broker` and the
+`secrets-provider-for-k8s` as they only consume libraries of the `conjur-authn-k8s-client`
+and will not be affected by the base image change. 
```

Comment:
how can you be sure? isn't it change the way the code is executed?

Reply:
The image of the authenticator-client is not built when we consume it in the secrets-provider so the base image doesn't affect it.

When you use goBoring as the base image it will affect how the code will run *on the authenticator image*, not anywhere that the code is consumed.